ultradns-rest/src/main/java/denominator/ultradns/UltraDNSRestGeoSupport.java

```diff
@@ -0,0 +1,242 @@
+package denominator.ultradns;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+import javax.inject.Named;
+
+import dagger.Module;
+import dagger.Provides;
+import denominator.ultradns.model.DirectionalGroup;
+import denominator.ultradns.model.Region;
+import denominator.ultradns.util.RRSetUtil;
+import denominator.ultradns.util.RegionUtil;
+import org.apache.commons.lang.StringUtils;
+
+@Module(injects = UltraDNSRestGeoResourceRecordSetApi.Factory.class, complete = false)
+public class UltraDNSRestGeoSupport {
+
+  private UltraDNSRest api = null;
+
+  static final int MAX_EFFECTIVE_CODES_PER_API_CALL = 100;
+
+  public UltraDNSRestGeoSupport() { }
+
+  public UltraDNSRestGeoSupport(UltraDNSRest api) {
+    this.api = api;
+  }
+
+  @Provides
+  @Named("geo")
+  Map<String, Collection<String>> regions(UltraDNSRest api1) {
```

Comment:
Why is this parameter required, when we already have the same instance variable available?